src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests_AwaitCompletion.vb

```diff
@@ -386,6 +424,392 @@ public class C
+                {
+                "await Task.Run(ASYNC () => someTask.$$);",
+                "await Task.Run(async () => await someTask$$);"
+                },
+                {
+                "await Task.Run(ASYNC () => someTask.aw$$);",
+                "await Task.Run(async () => await someTask$$);"
+                },
+                {
+                "await Task.Run(ASYNC () => {someTask.$$}",
+                "await Task.Run(async () => {await someTask$$}"
+                },
+                {
+                "await Task.Run(ASYNC () => {someTask.$$});",
+                "await Task.Run(async () => {await someTask$$});"
+                },
+                {
+                "await Task.Run(ASYNC () => someTask.   $$  );",
+                "await Task.Run(async () => await someTask$$  );"
+                },
+                {
+                "await Task.Run(ASYNC () => someTask  .   $$    );",
+                "await Task.Run(async () => await someTask  $$    );"
+                },
+                {
+                "await Task.Run(ASYNC () => someTask.$$.);",
+                "await Task.Run(async () => await someTask$$.);"
+                },
+                {
+                "Task.Run(async () => await someTask).$$",
+                "await Task.Run(async () => await someTask)$$"
+                }
+                )> expressionAndCommitted As String(), makeAsync As Boolean, appendConfigureAwait As Boolean) As Task
+
+            Dim expression = expressionAndCommitted(0).Replace("ASYNC ", If(makeAsync, "", "async "))
+            Using state = TestStateFactory.CreateCSharpTestState(
+<Document>
+using System.Threading.Tasks;
+
+static class Program
+{
+    static async Task Main()
+    {
+        var someTask = Task.CompletedTask;
+        <%= expression %>
+    }
+}
+</Document>)
+                state.SendInvokeCompletionList()
+                Await state.AssertCompletionItemsContainAll("await", "awaitf")
+                state.SendTypeChars("a")
+                state.SendTypeChars(If(appendConfigureAwait, "f", "w"))
+                Await state.AssertSelectedCompletionItem(If(appendConfigureAwait, "awaitf", "await"))
+
+                state.SendTab()
+                Dim committedTemplate = expressionAndCommitted(1)
+                Dim committed = committedTemplate.Replace("$$", If(appendConfigureAwait, ".ConfigureAwait(false)$$", "$$"))
+                Dim committedCursorPosition = committed.IndexOf("$$")
+                committed = committed.Replace("$$", "")
+                Assert.Equal($"
+using System.Threading.Tasks;
+
+static class Program
+{{
+    static async Task Main()
+    {{
+        var someTask = Task.CompletedTask;
+        {committed}
+    }}
+}}
+", state.GetDocumentText())
+                Await state.AssertLineTextAroundCaret($"        {committed.Substring(0, committedCursorPosition)}", $"{committed.Substring(committedCursorPosition)}")
+            End Using
+        End Function
```

Comment:
honestly, i can't read these tests and understand what they're doing.  this is why i'm not a fan of theories that do crazy stuff.  it just means that i spend an excess amount of time examining the code to support the theory.

Reply:
Understood (I felt the same, and that was the reason this test wasn't ported to VB yet). The problem is, that these are (4x11=44) single tests. Converting this to single test cases is a huge amount of work, with lots of duplication. I could drop the  `makeAsync` parameter which would simplify the test code and would remove the "ASYNC". But I would need to keep the handling of $$ in the expected code, because the test class  does not support checking the cursor position after commit with $$. This needs to be done with `state.AssertLineTextAroundCaret`.

Reply:
I changed the test in 5c7c1ed5 and the subsequent 2de160cc:
* Replaced `<CombinatorialData>` with `<InlineData>`
* Removed the two boolean parameters, that created the four test cases per "markup"
* Made the "Make container async" cases separate test cases (InlineData)
* Test await and awaitf completion within in the test

This brings the test `DotAwaitCompletionAddsAwaitInFrontOfExpressionInLambdas` in line with the existing `DotAwaitCompletionAddsAwaitInFrontOfExpressionForDifferentExpressions` which does about the same.

The number of tested cases is the same as before (21 InlineData and two tests (await and awaitf) in the test body).

The test is better readable than before, while still being compact and maintainable, IMHO.